nucleus/common/internal-api/src/test/java/org/glassfish/internal/deployment/GetApplicationNameTest.java

```diff
@@ -0,0 +1,124 @@
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package org.glassfish.internal.deployment;
+
+import java.io.IOException;
+import org.glassfish.api.deployment.DeploymentContext;
+import org.glassfish.api.deployment.archive.ReadableArchive;
+import org.glassfish.hk2.api.ServiceLocator;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ *
+ * @author Susan Rai
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class GetApplicationNameTest {
+
+    private static final String USER_PROVIDED_APP_NAME = "originalAppName";
+    //Mocks application name from reading the standard deployment descriptor.
+    private static final String DD_APP_NAME = "DDAppName";
+    private static final String DEFAULT_ARCHIVE_NAME = "defaultArchiveName";
+
+    @InjectMocks
+    private GenericHandlerStub genericHandler = new GenericHandlerStub();
+
+    @Mock
+    private ServiceLocator habitat;
+
+    @Mock
+    private ApplicationInfoProvider applicationInfoProvider;
+
+    private ReadableArchive readableArchive;
+
+    @Before
+    public void initialiseMocks() {
+        readableArchive = mock(ReadableArchive.class);
+        when(habitat.getService(ApplicationInfoProvider.class)).thenReturn(applicationInfoProvider);
+    }
+
+    @Test
+    public void testWhenDDHasAppName() {
+        when(applicationInfoProvider.getNameFor(readableArchive, null)).thenReturn(DD_APP_NAME);
+        assertEquals(DD_APP_NAME, genericHandler.getDefaultApplicationName(readableArchive, null, USER_PROVIDED_APP_NAME));
+    }
+
+    @Test
+    public void testWhenDDDoesNotHaveAppName() {
+        when(applicationInfoProvider.getNameFor(readableArchive, null)).thenReturn(null);
+        assertEquals(USER_PROVIDED_APP_NAME, genericHandler.getDefaultApplicationName(readableArchive, null, USER_PROVIDED_APP_NAME));
+    }
+
+    @Test
+    public void testWhenDDAppNameIsSetToDefaultArchiveName() {
+        when(applicationInfoProvider.getNameFor(readableArchive, null)).thenReturn(DEFAULT_ARCHIVE_NAME);
+        when(readableArchive.getName()).thenReturn(DEFAULT_ARCHIVE_NAME);
```

Comment:
I feel like a `assertSomething` helper method should be extended and used by each of the 3 test methods which should make them more readable in the the way: this input has that expected output. It should also become clear why only this text needs mockup of the `readableArchive.getName()`.